### PR TITLE
picobench: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **picobench/2.01**

picobench is a micro benchmarking library.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/picobench/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    @functools.lru_cache(1)
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["PICOBENCH_BUILD_TOOLS"] = self.options.with_cli
+        cmake.definitions["PICOBENCH_BIND_TO_ONE_CORE"] = self.options.bind_to_one_core
```
Comment (uilianries) [inline recipes/picobench/all/conanfile.py:54]: Weird, the option `PICOBENCH_BIND_TO_ONE_CORE` is present in the upstream's cmake, but there is no usage 🤔 

https://github.com/iboB/picobench/search?q=PICOBENCH_BIND_TO_ONE_CORE

It seems be leftover, or a future feature.

File: recipes/picobench/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 11)
+
+    def package_id(self):
+        if not self.options.with_cli:
+            self.info.header_only()
```
Comment (uilianries) [inline recipes/picobench/all/conanfile.py:40]:
```suggestion
            self.info.header_only()
        else:
            del self.info.settings.compiler
```
https://github.com/conan-io/conan-center-index/blob/master/docs/packaging_policy.md#settings

The docs is wrong, `del self.settings.compiler` only works under `configure` method.

File: recipes/picobench/all/conanfile.py (hunk 1 above)
Comment (toge) [reply recipes/picobench/all/conanfile.py]: Thank you for your review.
I missed it.
Now PICOBENCH_BIND_TO_ONE_CORE is deleted from recipe.

File: recipes/picobench/all/conanfile.py (hunk 2 above)
Comment (toge) [reply recipes/picobench/all/conanfile.py:40]: Fixed.

I didn't read this document. I will read this. 
Thank you.

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**